libs/go/sia/aws/agent/agent.go

```diff
@@ -91,11 +91,35 @@ func GetRoleCertificate(ztsUrl, svcKeyFile, svcCertFile string, opts *options.Op
 		return false
 	}
 
-	//initialize our return state to success
-	failures := 0
+	//fetch static roles
+	failures := fetchRoles(opts.Roles, opts, key, svcKeyFile, client)
```

Comment:
I'm somewhat confused with the lines 95-101. 

1. In the config we claim that //if true, fetch role certs dynamically. Roles attribute will be ignored. Based on the comments and this code we're fetching static roles and then dynamic roles.

2. when we address the fetch dynamic role part - why aren't we checking if the option is enabled in the first place? it looks like we always fetch the dynamic role certs.

Reply:
Fixed